cmd/kubelet/app/server.go

```diff
@@ -186,6 +186,120 @@ func Run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) error {
 	return nil
 }
 
+// setKubeDepsClients initializes clients if not standalone mode and any of the clients are not provided
+func setKubeDepsClients(s *options.KubeletServer, kubeDeps *kubelet.Dependencies, nodeName types.NodeName) error {
+	clientConfig, err := CreateAPIServerClientConfig(s)
+	if err == nil {
+		clientCertificateManager, err := makeKubeletClientCertificateManager(s, clientConfig, nodeName)
+		if err != nil {
+			return err
+		}
+
+		kubeClient, err := makeKubeDepsKubeClient(kubeDeps, clientConfig)
+		if err != nil {
+			startClientCertificateManager(kubeClient, clientCertificateManager)
```

Comment:
It's a little strange to require `startClientCertificateManager` to understand the api for `makeKubeletClientCertificateManager` (e.g. what to do with a nil client). I'd elevate those checks to `setKubeDepsClients`.

Reply:
ok